tool/scaffold/src/main/java/org/openscience/cdk/tools/scaffold/ScaffoldGenerator.java

```diff
@@ -0,0 +1,2412 @@
+/*
+ * Copyright (c) 2022 Julian Zander <zanderjulian@gmx.de>
+ *                    Jonas Schaub <jonas.schaub@uni-jena.de>
+ *                    Achim Zielesny <achim.zielesny@w-hs.de>
+ *                    Christoph Steinbeck <christoph.steinbeck@uni-jena.de>
+ *
+ * Contact: cdk-devel@lists.sourceforge.net
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package org.openscience.cdk.tools.scaffold;
+
+import org.openscience.cdk.CDKConstants;
+import org.openscience.cdk.aromaticity.Aromaticity;
+import org.openscience.cdk.aromaticity.ElectronDonation;
+import org.openscience.cdk.exception.CDKException;
+import org.openscience.cdk.fragment.MurckoFragmenter;
+import org.openscience.cdk.graph.ConnectivityChecker;
+import org.openscience.cdk.graph.CycleFinder;
+import org.openscience.cdk.graph.Cycles;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IAtomContainerSet;
+import org.openscience.cdk.interfaces.IAtomType;
+import org.openscience.cdk.interfaces.IBond;
+import org.openscience.cdk.interfaces.IChemObject;
+import org.openscience.cdk.interfaces.IRingSet;
+import org.openscience.cdk.interfaces.IStereoElement;
+import org.openscience.cdk.smiles.SmiFlavor;
+import org.openscience.cdk.smiles.SmilesGenerator;
+import org.openscience.cdk.tools.CDKHydrogenAdder;
+import org.openscience.cdk.tools.ILoggingTool;
+import org.openscience.cdk.tools.LoggingToolFactory;
+import org.openscience.cdk.tools.manipulator.AtomContainerManipulator;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.TreeMap;
+
+/**
+ * This class is designed to generate different molecule scaffolds and frameworks.
+ * It contains several other methods for the decomposition of molecules.
+ * All details of the functionality and implementation are described <a href="https://doi.org/10.26434/chemrxiv-2022-7tf0h-v2">
+ * here</a>.<p>
+ *
+ * Furthermore, the molecules can be decomposed according to the <a href="https://doi.org/10.1021/ci600338x">
```

Comment:
There is `@cdk.cite` for citing literature, but these needs revamping anyway. @johnmay, this is fine?

Reply:
Fine with me here

Reply:
Actually, thinking about https://github.com/cdk/cdk/issues/931 I just commented on, if I manage to factor out the JavaDoc taglets then we can do more. We can make the source code more fair, and make the citations available as machine readable code.

Reply:
Sorry, I'm a bit lost here. Should I use the @cdk.cite annotation or not (yet)? And if yes, how exactly do I do it?